packages/beacon-node/src/chain/chain.ts

```diff
@@ -112,6 +112,8 @@ export class BeaconChain implements IBeaconChain {
   // Seen cache for liveness checks
   readonly seenBlockAttesters = new SeenBlockAttesters();
 
+  readonly blockProposerIndices = new Map<Uint8Array, ValidatorIndex>();
```

Comment:
You shouldn't use Uint8Array as a Map key (except in certain special circumstances). Map uses object equality to match keys, so two distinct Uint8Arrays with the same binary contents will be treated as two distinct keys by the Map.

```ts
const m = new Map<Uint8Array, string>();
m.set(Uint8Array.from([0]), "foo");
m.get(Uint8Array.from([0])); // returns undefined
```

Reply:
Right, good catch